src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/google/GoogleComputeDAO.scala

```diff
@@ -0,0 +1,26 @@
+package org.broadinstitute.dsde.workbench.leonardo.dao.google
+
+import org.broadinstitute.dsde.workbench.leonardo.model.google._
+import org.broadinstitute.dsde.workbench.model.WorkbenchEmail
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+
+import scala.concurrent.Future
+
+/**
+  * Created by rtitle on 2/13/18.
+  */
+trait GoogleComputeDAO {
+
+  def getInstance(instanceKey: InstanceKey): Future[Option[Instance]]
+
+  def stopInstance(instanceKey: InstanceKey): Future[Unit]
+
+  def startInstance(instanceKey: InstanceKey): Future[Unit]
+
+  def addInstanceMetadata(instanceKey: InstanceKey, metadata: Map[String, String]): Future[Unit]
```

Comment:
`addInstanceMetadata` is used to install a startup script so Jupyter starts back up correctly on resume. Also it should help support https://github.com/DataBiosphere/leonardo/issues/128 (maybe we'll need a `removeInstanceMetadata` method).